db/views/patient_states_over_time_v01.sql

```diff
@@ -0,0 +1,69 @@
+SELECT
+    DISTINCT ON (p.id, month_date)
+    p.id,
+    p.recorded_at,
+    p.deleted_at,
+    cal.month,
+    cal.year,
+    cal.month_date,
+    bpot.systolic,
+    bpot.diastolic,
+    bpot.blood_pressure_recorded_at AS bp_recorded_at,
+    eot.encountered_at AS encountered_at,
+    p.assigned_facility_id AS patient_assigned_facility_id,
+    p.registration_facility_id AS patient_registration_facility_id,
+
+    (DATE_PART('year', cal.month_date) - DATE_PART('year', p.recorded_at)) * 12 +
+    (DATE_PART('month', cal.month_date) - DATE_PART('month', p.recorded_at))
+    AS months_since_registration,
+
+    CASE
+        WHEN mh.hypertension = 'yes' AND (bpot.systolic >= 180 OR bpot.diastolic >= 110) THEN 'Stage 3'
+        WHEN mh.hypertension = 'yes' AND (bpot.systolic >= 160 OR bpot.diastolic >= 100) THEN 'Stage 2'
+        WHEN mh.hypertension = 'yes' AND (bpot.systolic >= 140 OR bpot.diastolic >= 90) THEN 'Stage 1'
+        WHEN mh.hypertension = 'yes' AND (bpot.systolic < 140 AND bpot.diastolic < 90) THEN 'Controlled'
+        WHEN mh.hypertension = 'yes' AND (bpot.systolic IS null) THEN 'Hypertensive Unknown Stage'
+        WHEN mh.hypertension = 'unknown' THEN 'Unknown'
+        WHEN mh.hypertension = 'no' THEN 'Not hypertensive'
+        ELSE 'Undefined'
+        END
+        AS diagnosed_disease_state,
```

Comment:
Shall we capture `mh.hypertension` as a separate column as well, to make it easier to filter reports to "hypertensive only"? Seems more elegant than using these enum values everywhere.

Reply:
Done in de5aeb2fe4f5e7928d68fd2140606423dfcbac6f.